tests/integration/Products_Test.php

```diff
@@ -321,6 +322,116 @@ public function test_get_product_price_filter() {
+	 * @param bool $sync_enabled sync enabled for product
+	 * @param bool $expected_result the expected result
+	 *
+	 * @dataProvider provider_is_product_ready_for_commerce
+	 */
+	public function test_is_product_ready_for_commerce( $manage_stock_option, $manage_stock_prop, $product_price, $commerce_enabled, $sync_enabled, $expected_result ) {
+
+		$product = $this->get_product();
+
+		update_option( 'woocommerce_manage_stock', $manage_stock_option ? 'yes' : 'no' );
+		$product->set_manage_stock( $manage_stock_prop );
+		$product->set_regular_price( $product_price );
+		Products::update_commerce_enabled_for_product( $product, $commerce_enabled );
+		if ($sync_enabled) {
+			Products::enable_sync_for_products( [$product]);
+		} else {
+			Products::disable_sync_for_products( [$product]);
+		}
+
+		$this->assertEquals( $expected_result, Facebook\Products::is_product_ready_for_commerce( $product ) );
+	}
+
+
+	/** @see test_is_product_ready_for_commerce */
+	public function provider_is_product_ready_for_commerce() {
+
+		return [
+			[ true, true, '10.00', true, true, true ],
+			[ false, true, '10.00', true, true, false ],
+			[ true, false, '10.00', true, true, false ],
+			[ true, true, '0', true, true, false ],
+			[ true, true, '10.00', false, true, false ],
+			[ true, true, '10.00', true, false, false ],
+		];
+	}
+
+
+	/**
+	 * @see \SkyVerge\WooCommerce\Facebook\Products::is_commerce_enabled_for_product()
+	 *
+	 * @param string $meta_value meta value
+	 * @param bool $expected_result the expected result
+	 *
+	 * @dataProvider provider_is_commerce_enabled_for_product
+	 */
+	public function test_is_commerce_enabled_for_product( $meta_value, $expected_result ) {
+
+		$product = $this->get_product();
+
+		if ( ! empty( $meta_value ) ) {
+			$product->update_meta_data( Products::COMMERCE_ENABLED_META_KEY, $meta_value, true );
+		} else {
+			$product->delete_meta_data( Products::COMMERCE_ENABLED_META_KEY );
+		}
+
+		$this->assertEquals( $expected_result, Facebook\Products::is_commerce_enabled_for_product( $product ) );
+	}
+
+
+	/** @see test_is_commerce_enabled_for_product */
+	public function provider_is_commerce_enabled_for_product() {
+
+		return [
+			[ 'yes',  true ],
+			[ true,  true ],
+			[ 'no', false ],
+			[ false, false ],
+			[ null, false ], // if a product does not have this meta set, Commerce is not enabled for it
+		];
+	}
+
+
+	/**
+	 * @see \SkyVerge\WooCommerce\Facebook\Products::update_commerce_enabled_for_product()
+	 *
+	 * @param bool $param_value param value
+	 * @param string $expected_meta_value the expected meta value
+	 *
+	 * @dataProvider provider_update_commerce_enabled_for_product
+	 */
+	public function test_update_commerce_enabled_for_product( $param_value, $expected_meta_value ) {
+
+		$product = $this->get_product();
+
+		Products::update_commerce_enabled_for_product( $product, $param_value );
+
+		$this->assertEquals( $expected_meta_value, $product->get_meta( Products::COMMERCE_ENABLED_META_KEY ) );
```

Comment:
Mind updating this test to get a fresh instance of the product before asserting? Right now it doesn't test whether the meta is persisted.